src/generator/operations.ts

```
@@ -946,15 +942,15 @@ function generateResponseUnmarshaller(op: Operation, response: SchemaResponse, u
     // use the designated time type for unmarshalling
     unmarshallerText += `\tvar aux *${response.schema.language.go!.internalTimeType}\n`;
     unmarshallerText += `\tif err := runtime.UnmarshalAs${getMediaType(response.protocol)}(resp, &aux); err != nil {\n`;
-    unmarshallerText += `\t\treturn ${zeroValue}, runtime.NewResponseError(err, resp)\n`;
+    unmarshallerText += `\t\treturn ${zeroValue}, runtime.NewResponseError(resp)\n`;
```

This is incorrect as the unmarshalling error is lost.

---

This has been fixed.